test/unit/test_ranges.py

```diff
@@ -0,0 +1,310 @@
+                                               container_server)
+        self.object_server = eventlet.spawn(eventlet.wsgi.server,
+                                            self.obj_ring_socket,
+                                            object_server)
+        self.proxy_server = eventlet.spawn(eventlet.wsgi.server,
+                                           self.proxy_socket, proxy)
+
+        # Create test account
+        account = 'a'
+        partition, acc_nodes = proxy.account_ring.get_nodes(account)
+        acc_node = acc_nodes.pop()
+        acc_path = '/%s/%d/%s' % (acc_node['device'], partition, account)
+        acc_headers = {
+            'X-Timestamp': swift.common.utils.normalize_timestamp(time.time()),
+        }
+        acc_conn = httplib.HTTPConnection(acc_node['ip'], acc_node['port'])
+        acc_conn.request('PUT', acc_path, headers=acc_headers)
+        if acc_conn.getresponse().status != 201:
+            raise Exception("Unable to setup test account")
+
+        # Create test container
+        self.proxy_host, self.proxy_port = self.proxy_socket.getsockname()
+        proxy_conn = httplib.HTTPConnection(self.proxy_host, self.proxy_port)
+        proxy_conn.request('PUT', '/v1/a/c/')
+        if proxy_conn.getresponse().status != 201:
+            raise Exception("Unable to setup test container")
+
+    def tearDown(self):
+        # Reset monkey patched hash paths
+        swift.common.utils.HASH_PATH_SUFFIX = ''
+        swift.common.utils.HASH_PATH_PREFIX = ''
+
+        # Stop wsgi servers
+        self.account_server.kill()
+        self.container_server.kill()
+        self.object_server.kill()
+        self.proxy_server.kill()
+        self.sproxyd.kill()
+
+        # Close sockets
+        self.account_socket.close()
+        self.container_socket.close()
+        self.obj_ring_socket.close()
+        self.proxy_socket.close()
+        self.sproxyd_socket.close()
+
+        # Clean-up object rings
+        shutil.rmtree(self.swift_dir, ignore_errors=1)
+
+    def _get(self, path, headers):
+        """
+        Send a HTTP GET request to the swift test proxy.
+
+        :param path: Request path
+        :type path: str
+        :param headers: Request headers
+        :return: A tuple with response status, headers, and body.
+        """
+        proxy_conn = httplib.HTTPConnection(self.proxy_host, self.proxy_port)
+        proxy_conn.request('GET', path, headers=headers)
+        r = proxy_conn.getresponse()
+
+        return r.status, dict(r.getheaders()), r.read()
+
+    def test_range_from_byte(self):
+        self.response = '_' * 100
+
+        # Get last 95 bytes
+        headers = {'Range': 'bytes=5-'}
+        status, response_headers, _ = self._get('/v1/a/c/o', headers)
+        self.assertEqual(status, 206)
+        self.assertEqual(self.sproxyd_request_headers['RANGE'], 'bytes=5-99')
+        self.assertEqual(response_headers['content-range'], 'bytes 5-99/100')
```

Comment:
Could there be an assertion on the `Content-Length` header (here and in other tests)?